src/api/procedures/__tests__/modifyMultiSig.ts

```diff
@@ -330,7 +384,135 @@ describe('modifyMultiSig procedure', () => {
+      multiSig.details.mockResolvedValue({
+        signers: [oldSigner1, oldSigner2],
+        requiredSignatures: new BigNumber(3),
+      });
+
+      const proc = procedureMockUtils.getInstance<
+        ModifyMultiSigParams,
+        void,
+        ModifyMultiSigStorage
+      >(mockContext, {
+        signersToAdd: [],
+        signersToRemove: [],
+        requiredSignatures: new BigNumber(3),
+        currentSignerCount: 2,
+      });
+
+      const boundFunc = prepareStorage.bind(proc);
+
+      const result = await boundFunc({ signers: undefined, multiSig });
+      expect(result).toEqual({
+        signersToRemove: [],
+        signersToAdd: [],
+        requiredSignatures: new BigNumber(3),
+        currentSignerCount: 2,
+      });
+    });
+  });
+
+  describe('assertRequiredSignersExceedsSigners', () => {
+    it('should throw an error if the number of signatures required to be set exceeds the number of current signers', () => {
+      const expectedError = new PolymeshError({
+        code: ErrorCode.ValidationError,
+        message: 'The number of required signatures should not exceed the number of signers',
+      });
+
+      try {
+        assertRequiredSignersExceedsSigners(2, new BigNumber(2), undefined, new BigNumber(4));
+      } catch (error) {
+        expect(error).toEqual(expectedError);
+      }
+    });
+
+    it('should throw an error if the number of signers is less than required signatures', () => {
+      const expectedError = new PolymeshError({
+        code: ErrorCode.ValidationError,
+        message: 'The number of signers should not be less than the number of required signatures',
+      });
+
+      try {
+        assertRequiredSignersExceedsSigners(2, new BigNumber(3), [newSigner1, newSigner2]);
+      } catch (error) {
+        expect(error).toEqual(expectedError);
+      }
+    });
+
+    it('should throw an error if the number of signers is less than required signatures to be set', () => {
+      const expectedError = new PolymeshError({
+        code: ErrorCode.ValidationError,
+        message: 'The number of signers should not be less than the number of required signatures',
+      });
+
+      try {
+        assertRequiredSignersExceedsSigners(
+          2,
+          new BigNumber(2),
+          [newSigner1, newSigner2],
+          new BigNumber(3)
+        );
+      } catch (error) {
+        expect(error).toEqual(expectedError);
+      }
+    });
+  });
+
+  describe('assertNoDataChange', () => {
+    it('should throw an error if no signers are to be added/removed but were provided', () => {
+      const expectedError = new PolymeshError({
+        code: ErrorCode.NoDataChange,
+        message:
+          'The given signers are equal to the current signers. At least one signer should be added or removed',
       });
+
+      try {
+        assertNoDataChange(new BigNumber(2), [], [], [oldSigner1, oldSigner2]);
+      } catch (error) {
+        expect(error).toEqual(expectedError);
+      }
```

Comment:
```suggestion
        expect(() => assertNoDataChange(new BigNumber(2), [], [], [oldSigner1, oldSigner2])).toThrow(expectedError);
```

I think passing a function is cleaner syntax vs try catch.

Reply:
With catch syntax would the test fail if the assert function didn't throw?

It looks like the expect would be skipped and the test would still pass, so its not just a cosmetic change.

I think with try catch you need to assign to an error variable and assert outside the catch.